pkg/storage/stores/shipper/gateway_client.go

```diff
@@ -22,7 +23,10 @@ import (
 	util_math "github.com/grafana/loki/pkg/util/math"
 )
 
-const maxQueriesPerGoroutine = 100
+const (
+	maxQueriesPerGrpc      = 100
+	maxConcurrentGrpcCalls = 10
```

Comment:
Q: curious how do we come up with this value for `maxConcurrentGrpcCalls`?

Reply:
Using the benchmark that I wrote. I found that anything higher than 10 was not adding much value. I am going to push that benchmark in another PR with more improvements. We will see how it goes with this number. We can increase it if we find it too low for our prod envs.